enhancements/local-storage/health-monitoring.md

```diff
@@ -0,0 +1,155 @@
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes that we export health metrics for each local device on each node.
+
+## Motivation
+
+Local device vendors expose health information about storage devices via APIs like [SMART](https://en.wikipedia.org/wiki/S.M.A.R.T.). It very useful to know when a storage is about to fail, to differentiate between storage hardware related errors vs purely software ones, etc.
+
+### Goals
+
+- Expose health information about local storage devices in a way that can generate alerts
+
+### Non-Goals
+
+- Align with sig-storage [health monitoring for persistent volumes](https://kubernetes.io/blog/2021/04/16/volume-health-monitoring-alpha-update/)
+  - justification: it is also useful to monitor system drives and CSI only deals with persitent volumes, and migrating to CSI is an extremely long term effort.
+
+## Proposal
+
+[node_exporter](https://github.com/openshift/node_exporter) currently exports per-node metrics, we will use it to collect and export metrics for each device.
+
+### User Stories
+
+- As an admin, I would like to be alerted when a storage device is unhealthy and be linked to remediation steps.
+- As an admin, I would like if components (such as OpenShift Data Foundation) could access storage device health and failover faster.
+
+### Implementation Details/Notes/Constraints
+
+Implement a sidecar in the [node_exporter](https://github.com/openshift/node_exporter) component under [Cluster Monitoring Operator](https://github.com/openshift/cluster-monitoring-operator) that will scrape the metrics from the node and export them via the [textfile exporter](https://github.com/openshift/node_exporter/tree/master/text_collectors) (an exporter that scans `.prom` files in the
+`/var/node_exporter/textfile` directory and exports them.
+
+The Cluster Monitoring Operator will deploy the sidecar when the platform is Baremetal or None by modifying the `node_exporter` [daemonset](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/node-exporter/daemonset.yaml). The platform will be detected by querying the `infrastructure` resource.
+
+After that, the maintainance effort will mostly centered on the code the image runs to get health and update its `.prom` file. The logic for this collector can be in a subdirectory of CMO and have it's own OWNERS file allowing the ODF team to own it.
+
+We will be collecting metrics via [libstoragemgmt](https://github.com/libstorage/libstoragemgmt) and calling it via [libstoragemgmt-golang](https://github.com/libstorage/libstoragemgmt-golang/).
+If a block device is a partion or logical volume, we will only expose health for the parent device.
+
+
+### Ownership and Maintenance
```

Comment:
We need also to mention the service monitor (if we don't go the textfile collector's route) and the alerting rules. My idea is that they live in the cluster-monitoring-operator (because they need an operator to be deployed) but owned by the ODF team (again it should be feasible to put them in a dedicated directory with separate OWNERS file).